crates/core_simd/src/vector/float.rs

```diff
@@ -23,3 +23,24 @@ pub type f32x16 = SimdF32<16>;
 from_transmute_x86! { unsafe f32x4 => __m128 }
 from_transmute_x86! { unsafe f32x8 => __m256 }
 //from_transmute_x86! { unsafe f32x16 => __m512 }
+
+/// A SIMD vector of containing `LANES` `f64` values.
+#[repr(simd)]
+pub struct SimdF64<const LANES: usize>([f64; LANES])
+where
+    Self: crate::LanesAtMost64;
+
+impl_float_vector! { SimdF64, f64, SimdU64 }
```

Comment:
Perhaps the float macro should just be defined here?

Reply:
(and int, uint in their appropriate places as well)

Reply:
I moved the type aliases to the bottom of their respective files, the way everything works currently the type aliases are entangled with the code as-is. I will save that for a later refactoring.